plugins/wallabag.koplugin/main.lua

```diff
@@ -622,270 +692,457 @@ function Wallabag:download(article)
--- headers: defaults to auth if given nil value, provide all headers necessary if in use
--- body: empty string if not needed
--- filepath: downloads the file if provided, returns JSON otherwise
--- @treturn result or (nil, "network_error") or (nil, "json_error")
--- or (nil, "http_error", code)
----- @todo separate call to internal API from the download on external server
-function Wallabag:callAPI(method, apiurl, headers, body, filepath, quiet)
+
+    return skipped -- 2
+end                -- Wallabag:downloadArticle
+
+--- Call the wallabag API.
+-- See https://app.wallabag.it/api/doc/ for methods and parameters.
+-- @param method GET, POST, DELETE, PATCH, etc…
+-- @param url URL endpoint on wallabag server without hostname, or full URL for external link
+-- @param[opt] headers Defaults to Authorization for API endpoints, none for external
+-- @param[opt] body Body to include in the request, if needed
+-- @param[opt] filepath Downloads the file if provided, returns JSON otherwise
+-- @param[opt=false] quiet
+-- @treturn bool Whether the request was successful
+-- @treturn string Error type if unsuccessful, filepath if success with path, JSON if without
+-- @treturn int HTTP response code if unsuccessful (e.g. 404, 503, …)
+function Wallabag:callAPI(method, url, headers, body, filepath, quiet)
+    quiet = quiet or false
+
     local sink = {}
-    local request = {}
+    local request = { method = method }
 
     -- Is it an API call, or a regular file direct download?
-    if apiurl:sub(1, 1) == "/" then
+    --- @todo Separate call to internal API from the download on external server
+    if url:sub(1, 1) == "/" then
         -- API call to our server, has the form "/random/api/call"
-        request.url = self.server_url .. apiurl
-        if headers == nil then
-            headers = {
-                ["Authorization"] = "Bearer " .. self.access_token,
-            }
-        end
-    else
-        -- regular url link to a foreign server
-        local file_url = apiurl
-        request.url = file_url
-        if headers == nil then
-            -- no need for a token here
-            headers = {}
-        end
+        request.url = self.server_url .. url
+        request.headers = headers or {
+            ["Authorization"] = "Bearer " .. self.access_token,
+        }
+    else -- Assume full URL (e.g. https://…)
+        request.url = url
+        request.headers = headers or {}
     end
 
-    request.method = method
-    if filepath ~= "" then
+    if filepath ~= nil then
         request.sink = ltn12.sink.file(io.open(filepath, "w"))
         socketutil:set_timeout(socketutil.FILE_BLOCK_TIMEOUT, socketutil.FILE_TOTAL_TIMEOUT)
     else
         request.sink = ltn12.sink.table(sink)
         socketutil:set_timeout(socketutil.LARGE_BLOCK_TIMEOUT, socketutil.LARGE_TOTAL_TIMEOUT)
     end
-    request.headers = headers
-    if body ~= "" then
+
+    if body ~= nil then
```

Comment:
Same suspicious change.
```suggestion
    if body and body ~= "" then
```